Extract.hs

```diff
@@ -0,0 +1,28 @@
+import System.IO
+import System.Environment
+import Data.List
+
+main = do
+    args <- getArgs
+    header <- return( args !! 0)
+    file <- return( args !! 1)
+    -- putStrLn ("Searching for '" ++ header ++ "' in " ++ file)
+    withFile file ReadMode (\handle -> do
+        contents <- hGetContents handle
+        putStr $ processText header contents)
+
+processText :: String -> String -> String
```

Comment:
I would use `ByteString` for this, as it has an (IMO) richer api but is much more efficient too. It does add another dependency though.